constellations/gemini-admin/src/emailProgram/emailPrograms.js

```diff
@@ -0,0 +1,230 @@
+                    console.log("Error: api call failed. " + err.message);
+                });
+        } else {
+            this.setState({
+                usersInClass: [],
+                selectedUsers: [],
+            });
+        }
+    };
+
+    onCheckUser = (e, userId) => {
+        const users = this.state.usersInClass;
+        if (e.target.checked) {
+            var currentSelectedUsers = this.state.selectedUsers;
+            const checkedUser = users.find((user) => user.id == userId);
+            currentSelectedUsers.push(checkedUser);
+            this.setState({
+                selectedUsers: currentSelectedUsers,
+            });
+        } else {
+            var currentSelectedUsers = this.state.selectedUsers;
+            const uncheckedUser = users.find((user) => user.id == userId);
+            currentSelectedUsers.splice(
+                currentSelectedUsers.indexOf(uncheckedUser),
+                1
+            );
+            this.setState({
+                selectedUsers: currentSelectedUsers,
+            });
+        }
+    };
+
+    createUserRows = (users) => {
+        const userRows = users.map((user, index) => {
+            var status = user.isGuardian ? "Guardian" : "Student";
+            return (
+                <div className="row" key={index}>
+                    <input
+                        className="userSelect"
+                        type="checkbox"
+                        onChange={(e) => this.onCheckUser(e, user.id)}
+                    />
+                    <span className="column">{getFullName(user)}</span>
+                    <span className="column">{user.email}</span>
+                    <span className="column status">{status}</span>
+                </div>
+            );
+        });
+        return userRows;
+    };
+
+    render = () => {
+        const programOptions = this.state.programs.map((program, index) => (
+            <option key={index} value={program.programId}>
+                {program.name}
+            </option>
+        ));
+
+        const classOptions = this.state.classesForProgram.map(
+            (classes, index) => (
+                <div className="class" key={index}>
+                    <input
+                        type="checkbox"
+                        onChange={(e) => this.onCheckClass(e, classes)}
+                    />
+                    <span className="classId">{classes}</span>
+                </div>
+            )
+        );
+
+        const userRows = this.createUserRows(this.state.usersInClass);
+
+        return (
+            <div id="view-program-emails">
+                <section id="Select-program">
+                    <h1>Generate Email to Program</h1>
+
+                    <h2>Select a Program</h2>
+                    <select
+                        value={this.state.selectProgramId}
+                        onChange={(e) =>
+                            this.handleProgramChange(e, "selectProgramId")
```

Comment:
we needed the second argument in other pages because we wanted one function to handle multiple properties (like programId, programTitle, description, etc.). But here, we have a function to only handle changing the programId.

So this can just be `this.handleProgramChange(e)`
And the user-selected programId becomes e.target.value;